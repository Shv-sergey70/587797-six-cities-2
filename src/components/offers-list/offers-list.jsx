import React from 'react';
import PropTypes from 'prop-types';
import Card from "../card/card";
import offerPropTypes from '../../prop-types/offer';

const OffersList = (props) => {
  const {
    offers
  } = props;

  return <div className="cities__places-list places__list tabs__content">
    {offers.map((offer) => <Card
      key={`card-${offer.id}-${offer.isFavorite}`}
      offer={offer}
    />)}
  </div>;
};

OffersList.propTypes = {
  offers: PropTypes.arrayOf(offerPropTypes)
};

export {OffersList};
