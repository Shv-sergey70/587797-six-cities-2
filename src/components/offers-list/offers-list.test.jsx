import React from 'react';
import renderer from 'react-test-renderer';
import {OffersList} from "./offers-list";

const offersData = [
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: true,
    type: `Apartment`,
    price: 80,
    rating: 4.5
  },
  {
    id: 2,
    title: `Wood and stone place`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: false,
    type: `Private room`,
    price: 180,
    rating: 4
  },
  {
    id: 3,
    title: `Canal View Prinsengracht`,
    previewImage: `/img/apartment-02.jpg`,
    isPremium: false,
    isFavorite: true,
    type: `Apartment`,
    price: 120,
    rating: 5
  }
];

it(`OfferList correctly renders`, () => {
  const tree = renderer.create(
      <OffersList
        offers = {offersData}
        activeItem={0}
        changeActiveItem={jest.fn()}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
