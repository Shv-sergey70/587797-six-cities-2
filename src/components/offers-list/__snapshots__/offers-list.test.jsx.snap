// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferList correctly renders 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <Card
    offer={
      Object {
        "bedrooms": 2,
        "description": "A quiet cozy and picturesque that hides behind a a river",
        "goods": Array [
          "Heating",
          "Kitchen",
          "Cable TV",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
        ],
        "id": 1,
        "images": Array [
          "/img/room.jpg",
          "/img/apartment-01.jpg",
          "/img/studio-01.jpg",
        ],
        "isFavorite": true,
        "isPremium": true,
        "maxAdults": 2,
        "previewImage": "/img/apartment-01.jpg",
        "price": 80,
        "rating": 4.4,
        "title": "Beautiful & luxurious apartment at great location",
        "type": "apartment",
      }
    }
  />
  <Card
    offer={
      Object {
        "bedrooms": 5,
        "description": "A quiet cozy and picturesque that hides behind a a river1",
        "goods": Array [
          "Heating",
          "Kitchen",
          "Cable TV",
          "Washing machine",
          "Coffee machine",
        ],
        "id": 2,
        "images": Array [
          "/img/room.jpg",
          "/img/apartment-01.jpg",
          "/img/studio-01.jpg",
        ],
        "isFavorite": true,
        "isPremium": false,
        "maxAdults": 3,
        "previewImage": "/img/apartment-01.jpg",
        "price": 180,
        "rating": 2.4,
        "title": "Beautiful & luxurious apartment at great location2",
        "type": "apartment",
      }
    }
  />
  <Card
    offer={
      Object {
        "bedrooms": 5,
        "description": "A quiet cozy and picturesque that hides behind a a river22",
        "goods": Array [
          "Heating",
          "Kitchen",
          "Cable TV",
          "Washing machine",
        ],
        "id": 3,
        "images": Array [
          "/img/room.jpg",
          "/img/apartment-01.jpg",
        ],
        "isFavorite": false,
        "isPremium": true,
        "maxAdults": 4,
        "previewImage": "/img/apartment-01.jpg",
        "price": 820,
        "rating": 4.9,
        "title": "Beautiful & luxurious apartment at great location33",
        "type": "room",
      }
    }
  />
</div>
`;
